x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSender.java

```diff
@@ -515,15 +537,18 @@ Map<ShardId, List<DiscoveryNode>> resolveShards(Set<ShardId> shardIds) {
         var project = projectResolver.getProjectState(clusterService.state());
         var nodes = Maps.<ShardId, List<DiscoveryNode>>newMapWithExpectedSize(shardIds.size());
         for (var shardId : shardIds) {
-            nodes.put(
-                shardId,
-                project.routingTable()
+            List<DiscoveryNode> allocatedNodes;
+            try {
+                allocatedNodes = project.routingTable()
                     .shardRoutingTable(shardId)
                     .allShards()
                     .filter(shard -> shard.active() && shard.isSearchable())
                     .map(shard -> project.cluster().nodes().get(shard.currentNodeId()))
-                    .toList()
-            );
+                    .toList();
+            } catch (Exception ignored) {
+                continue;
+            }
```

Comment:
Should we continue on *any* exception here or just the one we get from the index being deleted. Is that just `IndexNotFoundException`?

Would you kindly add a comment around `continue;` saying "this is a shard being deleted after we started the request"?